modules/highgui/src/window_gtk.cpp

```diff
@@ -768,28 +776,18 @@ static Rect getImageRect_(const std::shared_ptr<CvWindow>& window)
 #ifdef HAVE_OPENGL
     if (window->useGl) {
         gtk_widget_translate_coordinates(window->widget, gtk_widget_get_toplevel(window->widget), 0, 0, &wx, &wy);
-        return Rect(wx, wy, window->widget->allocation.width, window->widget->allocation.height);
+        return Rect(wx, wy, gtk_widget_get_allocated_width(window->widget), gtk_widget_get_allocated_height(window->widget);
```

Comment:
```
/opencv/modules/highgui/src/window_gtk.cpp: In function 'cv::Rect getImageRect_(const std::shared_ptr<CvWindow>&)':
/opencv/modules/highgui/src/window_gtk.cpp:779:20: error: expected primary-expression before '(' token
  779 |         return Rect(wx, wy, gtk_widget_get_allocated_width(window->widget), gtk_widget_get_allocated_height(window->widget);
      |                    ^
/opencv/modules/highgui/src/window_gtk.cpp:779:124: error: expected ')' before ';' token
  779 |         return Rect(wx, wy, gtk_widget_get_allocated_width(window->widget), gtk_widget_get_allocated_height(window->widget);
      |                    ~                                                                                                       ^
```

```suggestion
        return Rect(wx, wy, gtk_widget_get_allocated_width(window->widget), gtk_widget_get_allocated_height(window->widget));
```

--------------------

To build with OpenGL support
* install packages: `libgtk2.0-dev libgtkglext1-dev`
* cmake options: `-DWITH_GTK=ON -DWITH_OPENGL=ON`